### DISCUSSION TITLE
Custom CSS using Nova::style on sail - file not found

### DISCUSSION THREAD

**Discussion by powellblyth** [laravel/nova-issues#3500] [Q&A]
- Laravel Version:8.53.0
- Nova Version: 3.28.0
- Sail Version: v1.9.0
- PHP Version:PHP 8.0.8 (cli) (built: Jul  1 2021 15:27:21) ( NTS )
- Database Driver & Version: MariaDB
- Operating System and Version: Sail on Mac (Intel)
- Browser type and version: Any
- Reproduction Repository: https://github.com/###/###

### Description:
I'm running Nova on Sail, and I wanted to add a custom button colour. 
I followed @davidhemphill 's video up to 1:20 to try and get _any_ css file to show
https://www.youtube.com/watch?v=InK-mRH9z5M&t=85s
However, http://localhost:6080/nova-api/styles/elasticutNova gives me 
```
ErrorException
file_get_contents(http://localhost:6080/css/elasticutNova.css): Failed to open stream: Cannot assign requested address
```
However, from my browser, I can access this URL `http://localhost:6080/css/elasticutNova.css` just fine

I think it's to do with PHP inside Sail trying to access the external address using file_get_contents, and failing, because it's obviously not a docker address

this is from `vendor/laravel/nova/src/Http/Controllers/StyleController.php:28`
```
public function show(NovaRequest $request)
    {
        $path = Arr::get(Nova::allStyles(), $request->style);
        abort_if(is_null($path), 404);
        return response(
            file_get_contents($path),
            200,
            [
                'Content-Type' => 'text/css',
            ]
        )->setLastModified(DateTime::createFromFormat('U', (string) filemtime($path)));
    }
```

**Comment by crynobone** [comment]
Please follow guide from our documentation, above video was release 2 years ago and using `asset()` with `file_get_contents()` would require you to enable `allow_url_fopen` 

https://www.php.net/manual/en/function.file-get-contents.php#refsect1-function.file-get-contents-notes
